task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -677,6 +685,34 @@ def startup() -> tuple[RuntimeTaskInstance, Context, Logger]:
         ti.log_url = get_log_url_from_ti(ti)
     log.debug("DAG file parsed", file=msg.dag_rel_path)
 
+    try:
+        run_as_user = getattr(ti.task, "run_as_user", None) or conf.get("core", "default_impersonation")
+    except AirflowConfigException:
+        run_as_user = None
+
+    if os.environ.get("_AIRFLOW__REEXECUTED_PROCESS") != "1" and run_as_user:
+        # enters here for re-exec process
+        os.environ["_AIRFLOW__REEXECUTED_PROCESS"] = "1"
+        # store startup message in environment for re-exec process
+        os.environ["_AIRFLOW__STARTUP_MSG"] = msg.model_dump_json()
+        os.set_inheritable(SUPERVISOR_COMMS.socket.fileno(), True)
+
+        # Import main directly from the module instead of re-executing the file.
+        # This ensures that when other parts modules import
+        # airflow.sdk.execution_time.task_runner, they get the same module instance
+        # with the properly initialized SUPERVISOR_COMMS global variable.
+        # If we re-executed the script, it would load as __main__ and future
+        # imports would get a fresh copy without the initialized globals.
+        rexec_python_code = "from airflow.sdk.execution_time.task_runner import main; main()"
+        log.info(
+            "Running command",
+            command=["sudo", "-E", "-H", "-u", run_as_user, sys.executable, "-c", rexec_python_code],
+        )
+        os.execvp("sudo", ["sudo", "-E", "-H", "-u", run_as_user, sys.executable, "-c", rexec_python_code])
```

Comment:
Nit: command is duplicated, we should store it in a variable used in both places.

Reply:
Handled